group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -1830,6 +1832,63 @@ public void onLoaded() {
         });
     }
 
+    /**
+     * Called when the partition is unloaded. Cancel all existing timers for the group.
+     * ClassicGroup: Complete all awaiting join futures and sync futures. Transition group to Dead.
+     */
+    public void onUnloaded() {
+        groups.values().forEach(group -> {
+            switch (group.type()) {
+                case CONSUMER:
+                    ConsumerGroup consumerGroup = (ConsumerGroup) group;
+                    log.info("[GroupId={}] Unloading group metadata for group epoch {}.",
+                        consumerGroup.groupId(), consumerGroup.groupEpoch());
+
+                    consumerGroup.members().values().forEach(member -> {
+                        timer.cancel(consumerGroupSessionTimeoutKey(consumerGroup.groupId(), member.memberId()));
+                        timer.cancel(consumerGroupRevocationTimeoutKey(consumerGroup.groupId(), member.memberId()));
```

Comment:
Cancelling the timers is not necessary here because all the timers of the shard are already cancelled when we reach this point. Check the unload method in the coordinator context.

Reply:
then we should expect classic group timers to also be cancelled already right?

Reply:
Yes.